nixos/modules/services/hardware/evdevremapkeys.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.evdevremapkeys;
+
+  fullConfig = {
+    devices = cfg.devices;
+  };
+
+  cfgFile = pkgs.writeText "config.yaml" (builtins.toJSON fullConfig);
+in {
+
+  options.services.evdevremapkeys = {
+
+    # Note: this corresponds to the YAML configuration that evdevremap keys
+    # expects.
+    devices = mkOption {
```

Comment:
Since the configuration file is `YAML` and we aren't really adding value by abstracting anything here it might make sense to simply provide a single `settings` option representing the file, then users can use a single attribute set to write the configuration file.

Reply:
I think we're adding value here: type safety. No?

Reply:
We could add an option `options.services.evdevremapkeys.extraConfig` of type `lines` 

and set  `config.services.evdevremapkeys.extraConfig = mkDefault cfgFile` ?

This is a pattern often used in NixOS to give people an escape hatch if nixos' module system gets out of sync with upstream temporarily...

However I don't really have a strong opinion.  this daemon's config seems small enough that I believe it will remain very stable for the forseeable future

Reply:
I'm not sure `extraConfig` provides value given we could use an attribute set to describe the file completely.

@q3k if you find value in the type checking the options provide please feel free to proceed with that.

Reply:
What about an `extraConfig` that is an attrset that gets merged into the typed config?

Reply:
If your options covers every possible file a user might want to generate by hand I think it is fine to omit `settings` and `extraConfig`, just going with your options.